docs/lib/project-setup/fragments/android/async/async.md

```diff
@@ -0,0 +1,72 @@
+Most functionalities in Amplify Android are exposed as asynchronous functions. These functions return immediately, returning a result at a later time:
+
+```kotlin
+val post = Post.builder()
+    .title("My First Post")
+    .build()
+
+// Time 0: save() is called.
+Amplify.DataStore.save(post,
+    {
+        // Time 2, later: a result or error is emitted
+        Log.i("MyAmplifyApp", "Saved a post")
+    },
+    {
+        // Time 2, later: a result of error is emitted
+        Log.e("MyAmplifyApp", "Save failed", it)
+    }
+)
+// Time 1: save() yields execution to following lines,
+// but no result available, yet.
+
+```
+
+This is a familiar pattern to many Android developers, and so is the default way of interacting with Amplify Android.
+
+However, this development model has some shortcomings. Let's say you need to wait for an operation to complete, so that you can perform additional logic that depends on its result. This can quickly become unmaintainable, resulting in a situation known as "callback hell":
```

Comment:
```suggestion
However, this development model has some challenges. Let's say you need to wait for an operation to complete, so that you can perform additional logic that depends on its result. This can quickly become unmaintainable, resulting in a situation known as "callback hell":
```